src/pip/_internal/utils/misc.py

```diff
@@ -548,7 +548,10 @@ def readline(self):
             try:
                 return next(self._gen)
             except NameError:
-                return self._gen.next()
+                # flake8-bugbear B305 suggests using six.next for
+                # Python 2 compatibility. This along with the try/except
+                # block can be removed once we drop Python 2 support
+                return six.next(self._gen)
```

Comment:
I wonder if we want to just import `next` from `six` and drop the `try`-`except` block *now*.

Reply:
Same here, I suspect the try-except block can be dropped entirely right now. `six.next()` should hadle that.

Reply:
Thanks, I have removed try/catch in https://github.com/pypa/pip/pull/8405/commits/47a975f1f557346dc21dff4565a268d89eae07aa

Reply:
The `next()` builtin was introduced in python 2.6. Do do we even need six here?

Reply:
> The `next()` builtin was introduced in python 2.6. Do do we even need six here?

You are right. The docs at https://docs.python.org/2/library/functions.html#next states the same. And since pip supports [Python 2 >=2.7.9](https://pip.pypa.io/en/latest/installing/#do-i-need-to-install-pip) it's a good idea to change it to just `next()`